### ISSUE TITLE
Cannot redefine property: propertyName

### ISSUE THREAD

**Issue by JamesIves** [angular/angular#42164]
# Bug Report

### Affected Package
`@angular/core`: 12.0.0
`zone.js`: 0.11.4

### Is this a regression?

Yes

### Description

After upgrading from Angular 11 -> 12 all tests which access `Object.defineProperty(obj, 'key', { value: 'false' })` now throw an error stating they cannot be redefined. This occurs if using defineProperty to override a module import or global level object. 

```typescript
afterEach(() => {
   Object.defineProperty(window.navigator, 'platform', {value: 'Win64'});
});
```

## Exception or Error

```
TypeError: Cannot redefine property: platform
	 at <Jasmine>
	 at UserContext.apply (src/app/shared/util.spec.ts:84:10)
	 at ZoneDelegate.invoke (node_modules/zone.js/fesm2015/zone.js:373:30)
	 at ProxyZoneSpec.onInvoke (node_modules/zone.js/fesm2015/zone-testing.js:291:43)
	 at ZoneDelegate.invoke (node_modules/zone.js/fesm2015/zone.js:372:56)
	 at Zone.run (node_modules/zone.js/fesm2015/zone.js:135:47)
	 at runInTestZone (node_modules/zone.js/fesm2015/zone-testing.js:571:38)
	 at UserContext.<anonymous> (node_modules/zone.js/fesm2015/zone-testing.js:586:24)
	 at <Jasmine>
```

## Your Environment

**Angular Version:**

```

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 12.0.0
Node: 12.18.4
Package Manager: npm 6.14.11
OS: darwin x64

Angular: 12.0.0
... animations, cdk, cli, common, compiler, compiler-cli, core
... forms, language-service, localize, material
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1200.0
@angular-devkit/build-angular   12.0.0
@angular-devkit/core            12.0.0
@angular-devkit/schematics      12.0.0
@schematics/angular             12.0.0
rxjs                            6.6.3
typescript                      4.2.4
    
```

**Comment by petebacondarwin** [comment]
It looks like something has already overridden this property but not set its `configurable` option to true.

**Comment by AndrewKushnir** [comment]
@JamesIves this might be related to the fix from #37582 which prevents ZoneJS from swallowing errors happened in `Object.defineProperty` callbacks. That fix was released in ZoneJS [v0.11.2](https://github.com/angular/angular/blob/master/packages/zone.js/CHANGELOG.md#0112-2020-09-19). You could try changing "zone.js" package version to `0.11.1` in `package.json` and see if the error is gone. If that's the case, there is a high chance that the error was there all the time, it was just hidden by the ZoneJS logic that patches `Object.defineProperty` function.

**Comment by JamesIves** [comment]
Sorry for the delayed response on this. I think you're right, it seemed like it was swallowing the error instead of showing it. I've adjusted my code and everything works as intended.
